WcaOnRails/app/webpacker/components/Auth/Signup/ClaimWcaId.jsx

```diff
@@ -0,0 +1,179 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Form, Message, FormGroup, FormField,
+} from 'semantic-ui-react';
+import DatePicker from 'react-datepicker';
+import { QueryClient, QueryClientProvider, useQuery } from '@tanstack/react-query';
+import WcaSearch from '../../SearchWidget/WcaSearch';
+import I18n from '../../../lib/i18n';
+import 'react-datepicker/dist/react-datepicker.css';
+import {
+  apiV0Urls, delegatesPageUrl, personUrl, contactDobUrl,
+} from '../../../lib/requests/routes.js.erb';
+import Loading from '../../Requests/Loading';
+import { fetchJsonOrError } from '../../../lib/requests/fetchWithAuthenticityToken';
+import I18nHTMLTranslate from '../../I18nHTMLTranslate';
+import UserBadge from '../../UserBadge';
+
+const dateFormat = 'YYYY-MM-DD';
+const queryClient = new QueryClient();
+
+function ClaimWcaId() {
+  const [unconfirmedPerson, setUnconfirmedPerson] = useState();
+  const [dobVerification, setDobVerification] = useState();
+  const [delegateToVerify, setDelegateToVerify] = useState();
+  const [manualDelegateSelection, setManualDelegateSelection] = useState(false);
+  const unconfirmedWcaId = unconfirmedPerson?.item?.wca_id || '';
+  const {
+    isLoading: isCanClaimLoading,
+    data: canClaimData,
+    refetch: refetchCanClaim,
+  } = useQuery(
+    {
+      queryKey: ['canClaim', unconfirmedWcaId],
+      queryFn: () => fetchJsonOrError(apiV0Urls.persons.canClaim(unconfirmedWcaId)),
+      enabled: !!unconfirmedWcaId,
+    },
+  );
+  const {
+    isLoading: isLikelyDelegatesLoading,
+    data: likelyDelegatesData,
+    refetch: refetchLikelyDelegates,
+  } = useQuery(
+    {
+      queryKey: ['likelyDelegates', unconfirmedWcaId],
+      queryFn: () => fetchJsonOrError(apiV0Urls.persons.likelyDelegates(unconfirmedWcaId)),
+      enabled: !!unconfirmedWcaId,
+    },
+  );
+  const { data: { can_claim: canClaim, reason } } = canClaimData || { data: {} };
+  const { data: { likely_delegates: likelyDelegates } } = likelyDelegatesData || { data: {} };
+  const loading = isCanClaimLoading || isLikelyDelegatesLoading;
+
+  useEffect(() => {
+    if (unconfirmedWcaId) {
+      refetchCanClaim();
+      refetchLikelyDelegates();
+    }
+  }, [refetchCanClaim, refetchLikelyDelegates, unconfirmedWcaId]);
+
+  useEffect(() => {
+    if (manualDelegateSelection) {
+      setDelegateToVerify(null);
+    }
+  }, [manualDelegateSelection]);
+
+  return (
+    // DatePicker inside semantic FormField works with proper CSS only if it's inside semantic Form.
+    // But currently it's inside rails form, hence to make the CSS work, <div className="ui form">
+    // is added instead of fragment.
+    <div className="ui form">
+      <Form.Field
+        control={WcaSearch}
+        label={I18n.t('common.user.wca_id')}
+        model="person"
+        multiple={false}
+        onChange={(_, { value }) => setUnconfirmedPerson(value)}
```

Comment:
`useInputState` hook to allow `onChange={setUnconfirmedPerson}`

Reply:
Done both